src/Admin/BlockAdmin.php

```diff
@@ -245,4 +214,69 @@ private function getDefaultTemplate(BlockServiceInterface $blockService)
             return $options['template'];
         }
     }
+
+    private function configureBlockFields(FormMapper $formMapper, BlockInterface $block): void
+    {
+        $service = $this->blockManager->get($block);
+        $blockType = $block->getType();
+
+        if (!$service instanceof BlockServiceInterface) {
+            throw new \RuntimeException(sprintf(
+                'The block "%s" is not a valid %s',
+                $block->getType(),
+                BlockServiceInterface::class
+            ));
+        }
+
+        if ($service instanceof EditableBlockService) {
+            $blockMapper = new PageFormMapper($formMapper);
+            if ($block->getId() > 0) {
```

Comment:
I don't know how these ids are set, but maybe a better check could be `null !== $block->getId()` if `0` can be considered a valid id.

Reply:
Not part of this PR. We will find some more type check errors when running phpstan